src/commands/purge-key.test.js

```diff
@@ -0,0 +1,101 @@
+const should = require(`should`);
+const sinon = require(`sinon`);
+require(`should-sinon`);
+
+const util = require(`./../util`);
+const testData = require(`./../test.data`)
+const testSubject = require(`./purge-key`);
+
+const testCacheKey = `5812587d-1407-4a60-930f-ce39e2b7e6f7`;
+
+// Scope vars that we’ll redefine on each test.
+let options, purgeKeyStub, softPurgeKeyStub, Fastly;
+
+// Begin testing.
+describe(`purge-key`, () => {
+  // Before and after each test, we reset some stuff
+  // to ensure a fresh set of arguments. See .beforeEach()
+  // and .afterEach() near the bottom of this file.
+
+  it(`should stop without an API key`, (done) => {
+    delete options.apikey;
+    testSubject(options, Fastly, util, testCacheKey);
+    util.apiKeyPresent.should.be.calledOnce();
+    Fastly.should.not.be.called();
+    done();
+  });
+
+  it(`should stop without a Service ID`, (done) => {
+    delete options.serviceid;
+    testSubject(options, Fastly, util, testCacheKey);
+    util.serviceIdPresent.should.be.calledOnce();
+    Fastly.should.not.be.called();
+    done();
+  });
+
```

Comment:
One good reason I can think of to check for these ourselves is consistent error reporting. Rather than have some errors produce a message with `fastly-cli` language and stack trace and others with a `fastly` language and stack trace, it would be less confusing to either always let `fastly` take care of reporting that or we always take care of that ourselves. I lean towards us handling it, but if the error messages from the `fastly` tool are useful/informative, then that might be the route to go.